packages/create-vite/src/index.ts

```diff
@@ -336,6 +336,9 @@ async function init() {
       })
 
     const [command, ...args] = fullCustomCommand.split(' ')
+    if (targetDir.includes(' ')) {
+      args.splice(2, args.length - 2, targetDir)
+    }
```

Comment:
I think simply replacing `args` instead of replacing the `fullCustomCommand` would be more easy to understand.
```js
// we replace TARGET_DIR here because targetDir may include a space
let replacedArgs = args.map(arg => arg.replace('TARGET_DIR', targetDir))
```

Would you also add a test here?
https://github.com/vitejs/vite/blob/main/packages/create-vite/__tests__/cli.spec.ts

Reply:
The code has been updated, but there seems to be something wrong with my device, could you add a test for me?

Reply:
When I add the test, I found that if `projectName` contains spaces, the value received by `progress.argv` will be split. Therefore, `targetDir` can only get the characters before the spaces. How to deal with this scense? 🤔

Reply:
In that case, I think the user should quote the project name (e.g. `npm init vite@latest "project name with spaces"`).

Reply:
Yes, However, when executing `pnpm create vite`, the input `Project name` may contain spaces. I think that after the template is created, if the project name contains spaces, the project name should be wrapped in quotes in the `cd projectName` output from the terminal.

Reply:
So you're suggest updating this line? I think that makes sense.
https://github.com/vitejs/vite/blob/697dd0085722eaa89697fd827ed90242c8040e0a/packages/create-vite/src/index.ts#L381

Reply:
Yes, I think it should be updated here. And the previous modification is also necessary. But I don't know how to write this test case?

Reply:
I tried writing the test but it seems to be quite complex. I think we can go without tests.

If we replace this `execaCommandSync` with `execaSync`, we can test the case of passing the project name from argv.
https://github.com/vitejs/vite/blob/2ff5930e02d80d6254037281b4c62b8e489d63ba/packages/create-vite/__tests__/cli.spec.ts#L16
But for the case of typing the project name, we need to feed the text from stdin and the test seems to hang.